specs/core/0_beacon-chain.md

```diff
@@ -1322,38 +1292,38 @@ def process_justification_and_finalization(state: BeaconState) -> None:
     # Process justifications
     state.previous_justified_epoch = state.current_justified_epoch
     state.previous_justified_root = state.current_justified_root
-    state.justification_bitfield = (state.justification_bitfield << 1) % 2**64
+    state.justification_bitfield = Bitvector[4](*([0b0] + state.justification_bitfield[0:JUSTIFICATION_BITVECTOR_LENGTH - 1]))
```

Comment:
The alternative to bitshift is in-place writes, similar to how `randao_mixes` is handled. It would avoid the `Bitvector[4](*(` magic, and be more consistent with the rest of the spec.

Reply:
You mean the modulo logic? I think in this case it would make the following code (that checks recently justified epochs and handles finalization) harder to read, so I would prefer the current state for readability if you don't have any other suggestions.